### PR TITLE
Add JSL Lexer

### PR DESCRIPTION
**Pull request by justinc11**
This PR adds support for JSL, the scripting language used in [JMP](www.jmp.com) software.

### CODE REVIEW

File: lib/rouge/lexers/jsl.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*- #
+
```
Comment (vidarh) [inline lib/rouge/lexers/jsl.rb]: Can you add '# frozen_string_literal: true'?

File: spec/lexers/jsl_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*- #
+
```
Comment (vidarh) [inline spec/lexers/jsl_spec.rb]: Can you add '# frozen_string_literal: true'?

**Comment by pyrmont** [comment]
@justinc11 I'm sorry it's taken so long to get back to you on this one. I also apologise for force pushing to your branch: I had a few comments on your code but it's been so long since you submitted it that I felt bad asking you to make changes to something you might not remember that well any more.

Hopefully all the changes are pretty straightforward. I've tried to make the lexing rules a little more robust based on a cursory reading of the JSL documentation. Please let me know if anything stands out as broken.

**Comment by justinc11** [comment]
Hi @pyrmont, thanks a bunch for cleaning this up for me (it needed some help). Since I initially opened this pull request I realized I forgot to account for date literals ([shown here](https://www.jmp.com/support/help/14-2/date-time-functions.shtml#221040)), but didn't get to it until your response reminded me. I also found out about a name-literal syntax similar to [SAS' name literals](https://documentation.sas.com/?docsetId=lrcon&docsetTarget=p18cdcs4v5wd2dn1q0x296d3qek6.htm&docsetVersion=9.4&locale=en#n1cqfh9044lm0fn1jq0vbmdut7c5) (e.g. `"some name!"n`) where you can have any characters in the name.

I pushed to a separate branch, [date-name-literal](https://github.com/justinc11/rouge/commit/33609509500a40889f4f489512b170f6ea0d81dd), rules for the date and name literals (I didn't want to mess with your already reviewed code). Could you let me know if these look okay or if any changes are needed before merging into this pull request?

**Comment by pyrmont** [comment]
@justinc11 Yep, that branch looks generally OK. Can you merge it in?

**Comment by justinc11** [comment]
Done. Thanks!

**Comment by pyrmont** [comment]
@justinc11 Thanks for the submission! Sorry again that it took so long but we got there in the end. The next release of Rouge is scheduled for this coming Tuesday so this lexer will be included in version 3.10.0 🎉

**Pull request merged**